### PR TITLE
ecl:  check that the range finder data accuracy

### PR DESCRIPTION
**Pull request by jiejiezhang**
check that the range finder data accuracy. Replace #472

### CODE REVIEW

File: EKF/terrain_estimator.cpp [hunk 1]
```diff
@@ -218,3 +221,10 @@ void Ekf::checkRangeDataContinuity()
 		_range_data_continuous = false;
 	}
 }
+
+// check that the range finder data accuracy
+void Ekf::checkRangeDataAccuracy()
+{
+	// get most recent range measurement from buffer
+	_rng_data_accuracy_good = _range_buffer.get_newest().variance <= 0.3;
```
Comment (mhkabir) [inline EKF/terrain_estimator.cpp]: Seems like an arbitrary magic number.

Comment (jiejiezhang) [reply EKF/terrain_estimator.cpp]: Yes, If you agree with this scheme, I will add it as a parameteｒ as soon.

Comment (jiejiezhang) [reply EKF/terrain_estimator.cpp]: @mhkabir  Done d6dfcd68fe93ebda7cd6ae235cc2ef9183d1a285

**Comment by priseborough** [comment]
I am trying to coordinate a number of changes and issues relating to range finder use at the moment:

https://github.com/PX4/Firmware/issues/9861
https://github.com/PX4/Firmware/pull/9886
https://github.com/PX4/Firmware/pull/9865

They will make this PR obsolete.

**Comment by priseborough** [comment]
@jiejiezhang PR https://github.com/PX4/Firmware/pull/9865 should meet your immediate requirement provided you modify your code so that publisher of the distance sensor topic sets the quality to 0 when the sensor specific checks fail.

Once https://github.com/PX4/Firmware/pull/9865 has been merged I will then submit additional PR's to move the checking the of the quality to inside the ecl library and also fix https://github.com/PX4/Firmware/issues/9861.

**Pull request closed without merging**